main.cpp

```diff
@@ -313,171 +345,153 @@ struct Staff
         void sickDays(int daysStatedInContract);
     };
     
-    // fix technical issues on board
     void fixTechnialIssuesOnBoard (int numOfStaffRequired);
-    // serve customers
     void serveCustomers();
-    // purchase with staff discount
-    void purchaseWithStaffDiscount(StaffWelfare discount);
+    void purchaseWithStaffDiscount(int discount);
 };
 
-/*
-Thing 6) BuildSpecification
-5 properties:
-    1) tons of steel used (double)
-    2) company materials were bought from (std::string)
-    3) volume of ship in m3 (float)
-    4) total weight in tons (double)
-    5) imported materials (std::string)
-3 things it can do:
-    1) describe material used
-    2) describe weight of ship
-    3) show building project date
- */
+int Staff::StaffWelfare::percentageOfTips(int customersServed, int hoursWorked)
+    {
+        customersServed = 20;
+        hoursWorked = 30;
+        int tips = customersServed / hoursWorked;
+        return tips;
+    }
+void Staff::StaffWelfare::sickDays(int daysStatedInContract)
+    {
+        daysStatedInContract = 10;
+    }
+void Staff::fixTechnialIssuesOnBoard (int numOfStaffRequired)
+    {
+        numOfStaffRequired = 5;
+    }
+void Staff::serveCustomers()
+    {
+    }
+void Staff::purchaseWithStaffDiscount(int discount)
+    {
+        discount = 30;
+    }
 
 struct BuildSpecification
 {
-    // tons of steel used
     double tonsOfSteelUsed = 23.59;
-    // company materials were bought from
     std::string companyMaterialsWereBoughtFrom = "Very good stuff";
-    // volume of ship in m3
     float volOfShipInMetreCube = 59834.34f;
-    // total weight in tons 
     double totalWeightInTons = 493.34;
-    // imported materials
     std::string importedMaterials = "aluminium";
 
-    // describe material used
     void describeMaterialUsed();
-    // describe weight of ship
     double displayWeightOfShip(double weightOfMaterials, int avgWeightOfPassengers, int maxCapacityOfPassengers);
-    // show building project date 
     void showBuildingProjectDate(std::string startDate, std::string endDate);
 };
-
-/*
-Thing 7) Entertainment
-5 properties:
-    1) name of act (std::string)
-    2) cost of hire (double)
-    3) allocated time (std::string)
-    4) number of performers (int)
-    5) location of performance (std::string)
-3 things it can do:
-    1) do acrobats
-    2) perform with audience
-    3) work for long haul
- */
+void::BuildSpecification::describeMaterialUsed()
+    {
+    }
+double::BuildSpecification::displayWeightOfShip(double weightOfMaterials, int avgWeightOfPassengers, int maxCapacityOfPassengers)
+    {
+        double weightOfShip = weightOfMaterials + avgWeightOfPassengers + maxCapacityOfPassengers;
+        return weightOfShip;
+    }
+void::BuildSpecification::showBuildingProjectDate(std::string startDate, std::string endDate)
+    {
+        startDate = "20th Feb 2000";
+        endDate = "31st Nov 2005";
+    }
```

Comment:
Fix this indentation
Don't write `returnType::` either (see comments below)
- [ ] revise